### PR TITLE
CSP-668 Reasons check your answers

### PR DESCRIPTION
**Pull request by YuvrajPatil-DFE**

### CODE REVIEW

**Review by DevenKShah — COMMENTED** [review]
I cannot review this story fully as it needs the changes from the other PR. I will wait till this is rebased.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs [hunk 1]
```diff
@@ -51,4 +51,26 @@ public class CheckYourAnswersControllerGetTests
...
+    OnboardingSessionModel sessionModel,
+    string reasonsUrl)
+    {
+        var reasonForJoiningTheNetwork = "The reason to join the network.";
+        var jobTitle = "Some Title";
+
+        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.Onboarding.ReasonToJoin, reasonsUrl);
+        sessionModel.ApprenticeDetails.ReasonForJoiningTheNetwork = reasonForJoiningTheNetwork;
+        sessionModel.ProfileData.Add(new ProfileModel { Id = ProfileDataId.JobTitle, Value = jobTitle });
+
```
[the remark this answers is not in the document]
Comment (YuvrajPatil-DFE) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs:68]: Noted

File: src/SFA.DAS.ApprenticeAan.Web/Models/Onboarding/CheckYourAnswersViewModel.cs [hunk 2]
```diff
@@ -12,12 +12,20 @@ public class CheckYourAnswersViewModel
...
+    public string ReasonForJoiningTheNetwork { get; }
+
     public CheckYourAnswersViewModel(IUrlHelper url, OnboardingSessionModel sessionModel)
     {
         JobTitleChangeLink = url.RouteUrl(@RouteNames.Onboarding.CurrentJobTitle)!;
         JobTitle = sessionModel.GetProfileValue(ProfileDataId.JobTitle)!;
 
         RegionChangeLink = url.RouteUrl(@RouteNames.Onboarding.Regions)!;
         Region = sessionModel.RegionName!;
+
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web/Models/Onboarding/CheckYourAnswersViewModel.cs:25]: remove space

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs [hunk 3]
```diff
@@ -52,6 +52,28 @@ public class CheckYourAnswersControllerGetTests
         result.As<ViewResult>().Model.As<CheckYourAnswersViewModel>().JobTitleChangeLink.Should().Be(TestConstants.DefaultUrl);
     }
 
+    [MoqAutoData]
+    public void Get_ReturnsViewResult_ValidReasons(
+        [Frozen] Mock<ISessionService> sessionServiceMock,
+        [Greedy] CheckYourAnswersController sut,
+        OnboardingSessionModel sessionModel,
+    string reasonsUrl)
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs:60]: Alignment is still off 😸

**Pull request merged**